tests/test_cluster/test_icdm.py

```diff
@@ -43,6 +43,8 @@
 ## Fixtures
 ##########################################################################
 
+Dataset = namedtuple('Dataset', 'X,y')
```

Comment:
I very much appreciate that you're trying to remove the `tests/dataset.py` file -- but this does lead to some code duplication; perhaps it would be better to put the `Dataset` and `Split` namedtuples into a `tests/fixtures.py` or `test/utils.py` for import?

Reply:
Good idea. I will put it in `tests/fixtures.py`. In the future, we can add other duplicated fixtures there~

Reply:
Ok, perfect thank you! Just one caution, anything that is in `conftest.py` is special and is available to all tests in that module without importing, see https://docs.pytest.org/en/latest/fixture.html#conftest-py-sharing-fixture-functions for more detail.